src/object_store/s3.rs

```diff
@@ -120,9 +120,13 @@ impl S3FileSystem {
 #[async_trait]
 impl ObjectStore for S3FileSystem {
     async fn list_file(&self, uri: &str) -> Result<FileMetaStream> {
-        let (_, prefix) = uri.split_once("s3://").ok_or_else(|| {
-            std::io::Error::new(ErrorKind::Other, S3Error::AWS("No s3 scheme found".into()))
-        })?;
+
+        let prefix = if let Some((_scheme, path)) = uri.split_once("://") {
```

Comment:
I wonder if there should be an assertion type check here for _scheme being s3?

Reply:
I was inclined to agree with this, but now I'm less sure. Should we really care what the scheme is?

I can imagine use-cases where the user registers not only s3 but also s3a and/or s3n as scheme to be used with this ObjectStore implementation...